### ISSUE TITLE
Development Guide tutorial

### ISSUE THREAD

**Issue by eak24** [AguaClara/aide_tutorial#10]
Hey @avtrigg, @prevosis I have just a few comments, but overall I really like how much this adheres to well-accepted practice. 
* The guide itself should be in aide_tutorial because it is not aide_design specific, and is therefore for startup and developer purposes. All things that apply to the whole AIDE ecosystem should go in aide_tutorial.
* This has some overlap with the standards and variable naming wiki pages. Perhaps those should be incorporated into this style guide. You may want to speak with @cc2467 in order to facilitate that move (and into aide_tutorial)
* It might be helpful to give summaries of the most important gotchyas from the linked style guides. I found it overwhelming to be linked over to a huge python style guide right off the bat. As an intimidated student, I think a little more handholding might be in order. 
* Is there some way we could make this more interactive and combine it with a series of tutorials modeled after the "First Contribution" tutorial? The goal here is to have a long-lasting collection of tutorials that can be used year after year. Could you have them develop a fun function that does something related to water treatment, like calculating the headloss in some system (river/building, etc...) and then you could do a group editing workshop to see what sort of style issues crop up and to test out the code reviewing system? Maybe the tutorials are modules in a getting started series. They could include a tutorial on "First Contribution" then "Code Style" and "Code Review." Do you think such a series would help the team get going? 
* The GUI should also have well-defined test cases or a test protocol that has various checks that can be done manually. 
* How should this system work with GitHub issues? (This probably shouldn't go in this page, but it's something to think about!) 

To summarize, my major thoughts are: try to go for more interactive with team review (this is fun, builds camaraderie and ensures everyone is on the same page), include fewer links to large docs and more short explanations that are AIDE specific - this is bound to grow as we move on. This tutorial repo should grow and grow :) - you're building this system to be used and upgraded year after year after year so any time you put towards tutorials will benefit the system for years to come!

**Comment by eak24** [comment]
Is there a reason the numpy docstring convention was recommended? I think I'm more a fan of the [Google convention,](http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) but I could certainly be persuaded!

**Comment by aadilzbhatti** [comment]
Both looked fine to us. We figured since a lot of the design code is using Numpy/Scipy/Matplotlib it might be a good idea to use the Numpy standard.
